src/types/compiler.rs

```diff
@@ -0,0 +1,18 @@
+use petgraph::stable_graph::StableDiGraph;
+
+use super::{contraption::Position, PowerLevel};
+
+pub enum NodeKind {}
+
+pub struct Node {
+    pos: Position,
+    power: PowerLevel,
+    kind: NodeKind,
+}
+
+pub enum Link {
+    // u8 being the distance till the next component
+    Power(u8),

```

Comment:
possibly need to specify the blocks that it exists in. Ie. if its 3, specify the coordinates of those 3 blocks. `Option<Vec<Position>>`?